Maze.cpp

```diff
@@ -114,7 +117,7 @@ Path Maze::findPath(CellCoordinate start, CellCoordinate end, Direction facing )
 			}
 
 			// This path is the best so far.
-			adjacentNode->previous = *currentNodeItr;
+			(*currentNodeItr)->next = adjacentNode;
```

Comment:
We could alternately store both; there shouldn't be a significant performance/memory cost to having both a `next` and `previous` pointer, right?